packages/@aws-cdk/cloud-assembly-schema/lib/protocol.ts

```diff
@@ -0,0 +1,60 @@
+import * as fs from 'fs';
+import { ArtifactMetadataEntryType, ArtifactType, AssemblyManifest } from './assembly-manifest';
+
+/**
+ * Protocol utility class.
+ */
+export class Manifest {
+
+    /**
+     * Save manifest to to file.
+     *
+     * @param manifest - manifest.
+     */
+    public static save(manifest: AssemblyManifest, filePath: string) {
+        fs.writeFileSync(filePath, JSON.stringify(manifest, undefined, 2));
+    }
+
+    /**
+     * Load manifest from file.
+     */
+    public static load(filePath: string): AssemblyManifest {
+        const manifest: AssemblyManifest = JSON.parse(fs.readFileSync(filePath, 'UTF-8'));
+        Manifest.patchStackTags(manifest);
+        return manifest;
+    }
+
+    /**
+     * This requires some explaining...
+     *
+     * We previously used `{ Key, Value }` for the object that represents a stack tag. (Notice the casing)
+     * @link https://github.com/aws/aws-cdk/blob/v1.27.0/packages/aws-cdk/lib/api/cxapp/stacks.ts#L427.
+     *
+     * When that object moved to this package, it had to be JSII compliant, which meant the property
+     * names must be `camelCased`, and not `PascalCased`. This meant it no longer matches the structure in the `manifest.json` file.
+     * In order to support current manifest files, we have to translate the `PascalCased` representation to the new `camelCased` one.
+     *
+     * Note that the serialization itself still writes `PascalCased` because it relates to how CloudFormation expects it.
+     *
+     * Ideally, we would start writing the `camelCased` and translate to how CloudFormation expects it when needed. But this requires nasty
+     * backwards-compatibility code and it just doesn't seem to be worth the effort.
+     */
+    private static patchStackTags(manifest: AssemblyManifest) {
```

Comment:
Is it possible to move this to the CLI or will something break?

Alternatively, can we just make this `any` and avoid all this mess? I don't feel comfortable about it.

Reply:
We can definitely move this to the CLI as its only called in the `load` method (notice it's private). Do you mean making the `manifest` argument `any`? Why?

Reply:
oh do you mean just get rid of the `Tag` interface and make it `any`?

Reply:
> We can definitely move this to the CLI

Scratch that, it needs to be called from both the CLI and the framework (`app.synth()`). So it either stays here or moves back to `cx-api`...

I keep falling in this trap because it doesn't feel natural that its invoked by the framework as well.

Reply:
> oh do you mean just get rid of the `Tag` interface and make it `any`?

Yes, seems like the pragmatic thing to do at the moment.

Reply:
I'm not entirely sure what the implications will be as far as the compatibility checks, because this would mean the union type i am using for the `MetadataEntry.data` property will now have `any[]`...I just need to make sure it doesnt break compat checks for other interfaces in the union. I'll give it a shot.

Reply:
@eladb I decided to leave this tags patching despite what we talked about.

I don't see any real downside to this, and it does provide us with more compatibility since we actually rely on this format. Doesn't feel right to change this to `any` just because the interface now moved to a `jsii` module.